packages/service-library/src/servicelib/fastapi/requests_decorators.py

```diff
@@ -119,4 +119,27 @@ async def wrapper(request: Request, *args, **kwargs):
     return wrapper
 
 
-__all__: tuple[str, ...] = ("cancel_on_disconnect",)
+def catch_n_raise(exception_types, http_status_map: Callable[[type], int]):
```

Comment:
I am not sure you really need that.
Please look here: https://github.com/ITISFoundation/osparc-simcore/blob/6e43b9feb093ae846ee7c441a1a4fdca06fde667/services/api-server/src/simcore_service_api_server/core/application.py

FastApi uses middlewares for such things. if there are specifics that you need that would be the correct way to implement that.